### PR TITLE
Add integration for DashVector (vector store)

### PR DESCRIPTION
**Pull request by xiaoyuxee**
# Description

This commit adds a VectorStore for DashVector vectorDB service.

[DashVector](https://help.aliyun.com/document_detail/2510225.html) is a fully-managed vectorDB service that supports high-dimension dense and sparse vectors, real-time insertion and filtered search. It is built to scale automatically and can adapt to different application requirements.

## Type of Change
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [ ] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: llama_index/vector_stores/dashvector.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+    def delete(self, ref_doc_id: str, **delete_kwargs: Any) -> None:
+        """
+        Delete nodes using with ref_doc_id.
+
+        Args:
+            ref_doc_id (str): The doc_id of the document to delete.
+
+        """
+        self._collection.delete(ids=ref_doc_id)
```
Comment (logan-markewich) [inline llama_index/vector_stores/dashvector.py:116]: Thanks for adding this to llamaindex! 🎉 

quick note -- the delete is actually intended to be a filtered delete

I.e. any node with `doc_id == ref_doc_id` in the metadata should be deleted.

Is this possible with dash vector?

Comment (xiaoyuxee) [reply llama_index/vector_stores/dashvector.py:116]: Thanks for your review! 🤝
oh, it is indeed a serious issue. But delete by filter will be support in DashVector soon.

so, I convert this pr to draft firstly

Comment (xiaoyuxee) [reply llama_index/vector_stores/dashvector.py]: > Thanks for your review! 🤝 oh, it is indeed a serious issue. But delete by filter will be support in DashVector soon.
> 

@logan-markewich The feature that delete by filter has released in DashVector. Can you do me a favor for the review? Thanks!

**Comment by Disiok** [comment]
Thanks so much for the contribution!

**Pull request merged**